Tests/DiligentCoreAPITest/src/ReadOnlyDepthTest.cpp

```diff
@@ -0,0 +1,614 @@
+ *  limitations under the License.
+ *
+ *  In no event and under no legal theory, whether in tort (including negligence),
+ *  contract, or otherwise, unless required by applicable law (such as deliberate
+ *  and grossly negligent acts) or agreed to in writing, shall any Contributor be
+ *  liable for any damages, including any direct, indirect, special, incidental,
+ *  or consequential damages of any character arising as a result of this License or
+ *  out of the use or inability to use the software (including but not limited to damages
+ *  for loss of goodwill, work stoppage, computer failure or malfunction, or any and
+ *  all other commercial damages or losses), even if such Contributor has been advised
+ *  of the possibility of such damages.
+ */
+
+#include "GPUTestingEnvironment.hpp"
+#include "TestingSwapChainBase.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace Diligent;
+using namespace Diligent::Testing;
+
+namespace
+{
+
+// clang-format off
+const std::string ReadOnlyDepthTest_VS{
+R"(
+struct PSInput
+{
+    float4 Pos : SV_POSITION;
+};
+
+struct VSInput
+{
+    float4 Pos : ATTRIB0;
+};
+
+void main(in  VSInput VSIn,
+          out PSInput PSIn)
+{
+    PSIn.Pos = VSIn.Pos;
+}
+)"
+};
+
+const std::string ReadOnlyDepthTest_PSDepth{
+R"(
+void main() {}
+)"
+};
+
+const std::string ReadOnlyDepthTest_PSColor{
+R"(
+Texture2D<float4> g_Input;
+
+float4 main() : SV_Target
+{
+    float depth = g_Input.Load(int3(0, 0, 0)).r;
+    return float4(depth, depth * 0.5, 0.75, 1.0);
+}
+)"
+};
+// clang-format on
+
+// Arbitrary clear color and depth that are never used anywhere else in this test
+constexpr float ClearColor[] = {0.11f, 0.32f, 0.02f, 1.0f};
+constexpr float ClearDepth   = 0.79f;
+
+constexpr float ExpectedDepth    = 0.5f;
+constexpr float ReferenceColor[] = {ExpectedDepth, ExpectedDepth * 0.5f, 0.75f, 1.0f};
+
+constexpr TEXTURE_FORMAT RTVFormat = TEX_FORMAT_RGBA8_UNORM;
+
+class ReadOnlyDepthTest : public testing::TestWithParam<std::tuple<int>>
+{
+protected:
+    static void SetUpTestSuite()
+    {
+        InitializeVertexBuffer();
+    }
+
+    static void TearDownTestSuite()
+    {
+        sm_pVB.Release();
+
+        auto* pEnv = GPUTestingEnvironment::GetInstance();
+        pEnv->Reset();
+    }
+
+    virtual void SetUp() override
+    {
+        InitializeDepthTexture();
+        InitializeRenderTarget();
+    }
+
+    virtual void TearDown() override
+    {
+        m_pDepthPSO.Release();
+        m_pColorPSO.Release();
+
+        m_pDepthTexture.Release();
+        m_pReadWriteDSV.Release();
+        m_pReadOnlyDSV.Release();
+
+        m_pRT.Release();
+        m_pRTV.Release();
+
+        m_pColorSRB.Release();
+    }
+
+    void SkipIfNotSupported()
+    {
+        auto* pEnv    = GPUTestingEnvironment::GetInstance();
+        auto* pDevice = pEnv->GetDevice();
+
+        const bool bSupported = !!(pDevice->GetTextureFormatInfoExt(GetDepthFormat()).BindFlags & BIND_DEPTH_STENCIL);
```

Comment:
Why do you need a double negation operation?

Reply:
To cast integer to bool without any warnings. I see that Diligent usually does (a & b) != 0 instead, I will change it for consistency.